en/middleware/micrortps.md

```diff
@@ -0,0 +1,344 @@
+    -r <reception port>     UDP port for receiving. Default 2019
+    -s <sending port>       UDP port for sending. Default 2020
+  ```
+  > **Note** If we are working with a USB-serial adapter the **-b** option will be ignored and will be working at maximum speed of the link.
+  ```sh
+  $ ./micrortps_agent # by default -t UART -d /dev/ttyACM0 -w 1 -b 460800 -p 1
+  ```
+
+Now we can add some code to print some info on the screen, for example:
+
+  ```cpp
+  void sensor_combined_Subscriber::SubListener::onNewDataMessage(Subscriber* sub)
+  {
+      // Take data
+      if(sub->takeNextData(&msg, &m_info))
+      {
+          if(m_info.sampleKind == ALIVE)
+          {
+              cout << "\n\n\n\n\n\n\n\n\n\n";
+              cout << "Received sensor_combined data" << endl;
+              cout << "=============================" << endl;
+              cout << "timestamp: " << msg.timestamp() << endl;
+              cout << "gyro_rad: " << msg.gyro_rad().at(0);
+              cout << ", " << msg.gyro_rad().at(1);
+              cout << ", " << msg.gyro_rad().at(2) << endl;
+              cout << "gyro_integral_dt: " << msg.gyro_integral_dt() << endl;
+              cout << "accelerometer_timestamp_relative: " << msg.accelerometer_timestamp_relative() << endl;
+              cout << "accelerometer_m_s2: " << msg.accelerometer_m_s2().at(0);
+              cout << ", " << msg.accelerometer_m_s2().at(1);
+              cout << ", " << msg.accelerometer_m_s2().at(2) << endl;
+              cout << "accelerometer_integral_dt: " << msg.accelerometer_integral_dt() << endl;
+              cout << "magnetometer_timestamp_relative: " << msg.magnetometer_timestamp_relative() << endl;
+              cout << "magnetometer_ga: " << msg.magnetometer_ga().at(0);
+              cout << ", " << msg.magnetometer_ga().at(1);
+              cout << ", " << msg.magnetometer_ga().at(2) << endl;
+              cout << "baro_timestamp_relative: " << msg.baro_timestamp_relative() << endl;
+              cout << "baro_alt_meter: " << msg.baro_alt_meter() << endl;
+              cout << "baro_temp_celcius: " << msg.baro_temp_celcius() << endl;
+
+              // Print your structure data here.
+              ++n_msg;
+              //std::cout << "Sample received, count=" << n_msg << std::endl;
+              has_msg = true;
+
+          }
+      }
+  }
+  ```
+
+> **Note** Normally, for UART transport it's necessary set up the UART port in the Raspberry Pi. To enable the serial port available on Raspberry Pi connector:
+
+1. Make sure the `userid` (default is pi) is a member of the `dialout` group:
+
+  ```sh
+  $ groups pi
+  $ sudo usermod -a -G dialout pi
+  ```
+
+2. You need to stop the already running on the GPIO serial console:
+
+  ```sh
+  $ sudo raspi-config
+  ```
+
+  Go to **Interfacing options > Serial**. Select **NO for *Would you like a login shell to be accessible over serial?*. Valid and reboot.
```

Comment:
I don't understand these instructions. Are they prompts I'll see somewhere in some UI?